### ISSUE TITLE
Provide functionality on your website to test a config and alert to deprecation

### ISSUE THREAD

**Issue by U-4-E-A** [stylelint/stylelint-demo#396]
### What is the problem you're trying to solve?

I recently updated from v14 to v16 and encountered a number of deprecated rules. stylelint's documentation is excellent but the "migrating to v15" simply leaves me having to check and remove individual rules. Whereas this is not the end of the world, I think it would be beneficial for stylelint to provide a view on their website which would automate cleaning up of config.

### What solution would you like to see?

A view where a user can paste their config into a text field, select their stylelint version (and perhaps also add in the same selection functionality for the various plugins) and be alerted to rules (and other parts of the config) that are deprecated or removed. Perhaps also allow the user to download a new version of the config with the removed rules deleted and maybe also download with comments added stating where a rule is deprecated and is expected to be removed.

**Comment by ybiquitous** [comment]
[This demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgdwEsocALeFARgAYAFADQG4sBfLEAGhADMCAbGAHJIAtnEQwAHiIAO-AHRgAzoo7gImHgHMEIDJhQp0IAE4BXfoqPk9Bg0dOYCOALRgkimFcMheEPDGMjLANWTGZVSA0CTQAxCGNhJBwdACtFdVVYaRVEG29FHABPfl4CTBwvIwoAViCwkGYgA) is an example of the `unit-case` rule that was removed in 16.0.0. No warnings are shown:

<img width="467" alt="image" src="https://github.com/stylelint/stylelint-demo/assets/473530/b580669c-1841-45b6-9cc3-aa53cdafd657">

On the other hand, the deprecation warning is output with CLI:

```sh-session
$ npx stylelint -v
15.11.0

$ echo 'a { width: 1PX; }' | npx stylelint

Deprecation warnings:
 - The "unit-case" rule is deprecated.

<input css RjcOe8>
 1:13  ✖  Expected "PX" to be "px"  unit-case

1 problem (1 error, 0 warnings)
```

**Comment by Mouvedia** [comment]
> select their stylelint version […] and be alerted to rules […] that are deprecated¹ or removed²

If you have `"stylelint": "latest",` in the dependencies tabs, removed rules will get the same treatment as typos or third party rules lacking their plugins.
i.e. `Unknown rule foo-bar.`

So there's 2 part to this:

1. showing the deprecation warnings
2. adding [`deprecated`](https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-00#rfc.section.9.3) rules of the last major to stylelintrc.json so that they can be detected as such (instead of being treated as unknown)
